### DISCUSSION TITLE
Grouping of Input Maps

### DISCUSSION THREAD

**Discussion by AdminCrystal** [godotengine/godot-proposals#5278] [Editor]
Some games are going to have a lot of Controls being put into the input map. It would make organizing and visualization better if we could create groups inside the Input Map section of Project Settings.

For example lets say I have an open world game that has cars, boats and helicopters. I want to have separate controls for each type of vehicle and a set of controls for the player and a different set for Menus.

Currently:
I would add all these controls inside the input map and organize them so they are all in order to make it easier for myself. Now I have say 50 inputs all in one area and if I want to modify them I have to scroll and find the specific one I want to modify

Proposal:
Create groupings so I can put all helicopter controls in one group etc. Also an option to expand and minimize everything inside that group so I don't have to see all the controls I am not concerned with at that time.

Bonus but less important proposal:
Be able to turn on and off groups of controls. So for example if someone is inside a helicopter I can turn off all those inputs inside of other groups such as the boats and car groups.

Here is an example. Pretend the keycodes are all different controls with their own unique controls beneath them
Open groups
![image](https://user-images.githubusercontent.com/50718608/187052995-caf92e4b-8914-4400-9d73-46a5ea66bf77.png)

Closed groups
![image](https://user-images.githubusercontent.com/50718608/187053022-86114861-1a98-4b17-a826-4086138848ea.png)

**Comment by Calinou** [comment]
This is already being tracked in https://github.com/godotengine/godot-proposals/issues/4910.
